### PR TITLE
[docs] More API fixes

### PR DESCRIPTION
**Pull request by stevhliu**
Adds the following API fixes:

- Return statements to the `download`, `from_config`, and `load_config` functions
- Move `ImagePipelineOutput`, `AudioPipelineOutput`, and `ImageTextPipelineOutput` to the Outputs page
- Moves non "Stable Diffusion" pipelines out of the section (included `SafeStableDiffusion` here)
- Create new Utilities API page for `randn_tensor`
- Clarify parameters in docstring

### CODE REVIEW

File: docs/source/en/api/diffusion_pipeline.mdx [hunk 1]
```diff
@@ -12,19 +12,19 @@ specific language governing permissions and limitations under the License.
...
 
-The [`DiffusionPipeline`] is the easiest way to load any pretrained diffusion pipeline from the [Hub](https://huggingface.co/models?library=diffusers) and to use it in inference.
+The [`DiffusionPipeline`] is the easiest way to load any pretrained diffusion pipeline from the [Hub](https://huggingface.co/models?library=diffusers) and use it for inference.
 
 <Tip>
 	
-	One should not use the Diffusion Pipeline class for training or fine-tuning a diffusion model. Individual 
-	components of diffusion pipelines are usually trained individually, so we suggest to directly work 
-	with [`UNetModel`] and [`UNetConditionModel`].
+	You shouldn't use the [`DiffusionPipeline`] class for training or finetuning a diffusion model. Individual
```
Comment (sayakpaul) [inline docs/source/en/api/diffusion_pipeline.mdx:19]:
```suggestion
	You shouldn't use the [`DiffusionPipeline`] class for training or finetuning a diffusion 
model. Individual 
	components ([`UNet1DModel`], [`UNet2DConditionModel`], for example) of diffusion pipelines are usually trained individually, so we suggest directly working with them instead.
```

Comment (sayakpaul) [reply docs/source/en/api/diffusion_pipeline.mdx:19]: The suggestion might be broken. Proceed with caution.

File: docs/source/en/api/diffusion_pipeline.mdx [hunk 2]
```diff
@@ -35,18 +35,3 @@ Any pipeline object can be saved locally with [`~DiffusionPipeline.save_pretrain
...
-
-## AudioPipelineOutput
-By default diffusion pipelines return an object of class
-
-[[autodoc]] pipelines.AudioPipelineOutput
-
-## ImageTextPipelineOutput
-By default diffusion pipelines return an object of class
-
-[[autodoc]] ImageTextPipelineOutput
```
Comment (sayakpaul) [inline docs/source/en/api/diffusion_pipeline.mdx:52]: Where are these going?

Comment (sayakpaul) [reply docs/source/en/api/diffusion_pipeline.mdx:52]: Ah nevermind.

File: docs/source/en/api/outputs.mdx [hunk 3]
```diff
@@ -25,31 +25,35 @@ pipeline = DDIMPipeline.from_pretrained("google/ddpm-cifar10-32")
...
+
+[[autodoc]] pipelines.ImagePipelineOutput
+
+## AudioPipelineOutput
+
+[[autodoc]] pipelines.AudioPipelineOutput
+
+## ImageTextPipelineOutput
+
+[[autodoc]] ImageTextPipelineOutput
```
Comment (sayakpaul) [inline docs/source/en/api/outputs.mdx:59]: We plan to add ALL the output classes here, yeah?

File: docs/source/en/api/utilities.mdx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Utilities
+
+Utility and helper functions for working with 🤗 Diffusers.
+
+## randn_tensor
+
+[[autodoc]] diffusers.utils.randn_tensor
```
Comment (sayakpaul) [inline docs/source/en/api/utilities.mdx:7]: Might be worth including the following too:

* numpy_to_pil
* pt_to_pil
* load_image
* export_to_video


I know there are quite a few [here](https://github.com/huggingface/diffusers/blob/main/src/diffusers/utils/__init__.py), I only picked the ones I use the most. 

WDYT?

**Review by sayakpaul — APPROVED** [review]
Magician there 🪄

**Review by patrickvonplaten — APPROVED** [review]
Very cool!

File: docs/source/en/api/outputs.mdx (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/api/outputs.mdx:59]: I was thinking we just add these output types and then leave the pipeline/model specific output classes (for example, `AltDiffusionPipelineOutput`) in their respective [pages](https://huggingface.co/docs/diffusers/main/en/api/pipelines/alt_diffusion). Maybe we can add a tip to check the individual pipeline/model pages for it?

File: docs/source/en/api/utilities.mdx (hunk 4 above)
Comment (stevhliu) [reply docs/source/en/api/utilities.mdx:7]: That sounds good! For starters I'll add the ones you listed above, and if there are other utilities we see that get a lot of usage we can add them to the page.

File: docs/source/en/api/outputs.mdx (hunk 3 above)
Comment (sayakpaul) [reply docs/source/en/api/outputs.mdx]: I think that's better. We provide an example link with your comment. Let's incorporate this change and ship this 🚢

File: docs/source/en/api/diffusion_pipeline.mdx [hunk 5]
```diff
@@ -12,19 +12,18 @@ specific language governing permissions and limitations under the License.
...
+components (for example, [`UNetModel`] and [`UNetConditionModel`]) of diffusion pipelines are usually trained individually, so we suggest directly working with instead.
 
 </Tip>
 
-Any diffusion pipeline that is loaded with [`~DiffusionPipeline.from_pretrained`] will automatically 
-detect the pipeline type, *e.g.* [`StableDiffusionPipeline`] and consequently load each component of the 
-pipeline and pass them into the `__init__` function of the pipeline, *e.g.* [`~StableDiffusionPipeline.__init__`].
+The pipeline type (for example [`StableDiffusionPipeline`]) of any diffusion pipeline loaded with [`~DiffusionPipeline.from_pretrained`] is automatically 
+detected and consequently loads each component of the 
+pipeline and passes them into the `__init__` function of the pipeline.
```
Comment (pcuenca) [inline docs/source/en/api/diffusion_pipeline.mdx:26]:
```suggestion
The pipeline type (for example [`StableDiffusionPipeline`]) of any diffusion pipeline loaded with [`~DiffusionPipeline.from_pretrained`] is automatically 
detected and pipeline components are loaded and passed to the `__init__` function of the pipeline.
```

File: docs/source/en/api/outputs.mdx [hunk 6]
```diff
@@ -12,11 +12,11 @@ specific language governing permissions and limitations under the License.
 
 # BaseOutputs
 
-All models have outputs that are instances of subclasses of [`~utils.BaseOutput`]. Those are
-data structures containing all the information returned by the model, but that can also be used as tuples or
+All models have outputs that are subclass instances of [`~utils.BaseOutput`]. Those are
```
Comment (pcuenca) [inline docs/source/en/api/outputs.mdx:15]:
```suggestion
All models have outputs that are subclasses of [`~utils.BaseOutput`]. Those are
```

File: docs/source/en/api/utilities.mdx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+## numpy_to_pil
+
+[[autodoc]] utils.pil_utils.numpy_to_pil
+
+## pt_to_pil
+
+[[autodoc]] utils.pil_utils.pt_to_pil
```
Comment (pcuenca) [inline docs/source/en/api/utilities.mdx:15]: Some of these are usually accessed as static functions of the pipelines, which is usually convenient. Not sure if we should cross-reference them somehow (not necessary for this PR, just to keep in mind).

File: src/diffusers/pipelines/pipeline_utils.py [hunk 8]
```diff
@@ -1109,95 +1109,78 @@ def load_module(name, value):
...
+                community pipeline](https://huggingface.co/docs/diffusers/main/en/using-diffusers/contribute_pipeline).
 
             force_download (`bool`, *optional*, defaults to `False`):
                 Whether or not to force the (re-)download of the model weights and configuration files, overriding the
                 cached versions if they exist.
             resume_download (`bool`, *optional*, defaults to `False`):
-                Whether or not to delete incompletely received files. Will attempt to resume the download if such a
-                file exists.
+                Whether or not to resume downloading the model weights and configuration files. If set to False, any
+                incompletely downloaded files are deleted.
```
Comment (pcuenca) [inline src/diffusers/pipelines/pipeline_utils.py:1148]: ❤️

**Review by pcuenca — APPROVED** [review]
Amazing work, thanks a lot!

**Pull request merged**